swarm/pss/types.go

```diff
@@ -91,29 +93,65 @@ type pssDigest [digestLength]byte
 
 // conceals bitwise operations on the control flags byte
 type msgParams struct {
-	raw bool
-	sym bool
+	raw bool // unenctypted mode
+	sym bool // symmetric encryption
+	nhr int  // neighbourhood addressing mode based on minumim proximity order
+	nhs int  // neighbourhood addressing mode based on minumim neighbourhood size
 }
 
 func newMsgParamsFromBytes(paramBytes []byte) *msgParams {
-	if len(paramBytes) != 1 {
+	if len(paramBytes) == 0 {
 		return nil
 	}
+	var nhs, nhr int
+	if paramBytes[0]&pssControlNeighbourhoodRadius > 0 {
+		if len(paramBytes) > 1 {
+			nhr = int(uint8(paramBytes[1]))
+		} else {
+			return nil
+		}
+		if paramBytes[0]&pssControlNeighbourhoodSize > 0 {
+			if len(paramBytes) > 2 {
+				nhs = int(uint8(paramBytes[2]))
+			} else {
+				return nil
+			}
+		}
+	} else if paramBytes[0]&pssControlNeighbourhoodSize > 0 {
+		if len(paramBytes) > 1 {
+			nhs = int(uint8(paramBytes[1]))
+		} else {
+			return nil
+		}
+	}
+
 	return &msgParams{
 		raw: paramBytes[0]&pssControlRaw > 0,
 		sym: paramBytes[0]&pssControlSym > 0,
+		nhs: nhs,
+		nhr: nhr,
```

Comment:
Why do we need both in the bitfield? Will one ever be set without the other?

Reply:
yes why not

Reply:
in `isSelfPossibleReceipient` nhr is ever only used if nhs is set.

Reply:
In which cases would you only specify S and partial `To` address for example?

Reply:
with S specified you can afford to have a partial address only. 
What i mean by obfuscated dark message is that sending to a n-byte partial address is equivalent to sending to a bogus full address with at least n-long full address and specifying R=8n. which implies that partial addressing can be eliminated really.

Reply:
cannot think of a use case for that. In fact if the partial TO address is shorter than the Depth needed for S then noone will pick up the message...